docs/content/python/getting-started/application/model.md

```diff
@@ -0,0 +1,214 @@
+    # "contributor" role.
+    "read" if "contributor";
+
+    # A user has the "push" permission if they have the
+    # "maintainer" role.
+    "push" if "maintainer";
+}
+```
+
+This policy declares `"read"` and `"push"` as permissions for the
+`Repository` resource and assigns them to roles. We also tell Oso
+that our `User` class is an `actor` with the `actor User {}`
+declaration.
+
+The `"read" if "contributor";` statement is an example of a *shorthand rule.*
+Add an `"admin"` role and give it its own permissions by adding some
+more shorthand rules:
+
+```polar
+resource Repository {
+    permissions = ["read", "push", "delete"];
+    roles = ["contributor", "maintainer", "admin"];
+
+    "read" if "contributor";
+    "push" if "maintainer";
+
+	# A user has the "delete" permission if they have the
+	# "admin" role.
+	"delete" if "admin";
+
+	# A user has the "maintainer" role if they have
+	# the "admin" role.
+    "maintainer" if "admin";
+
+	# A user has the "contributor" role if they have
+	# the "maintainer" role.
+    "contributor" if "maintainer";
+}
+```
+
+The last rules we added are between two roles: A user has the
+`"maintainer"` role and all permissions associated with it if they have the
+`"admin"` role, and the same for `"contributor"` and `"maintainer"`.
+
+### Giving your users roles
+
+Now that we've finished our policy, we must associate users with roles
+in our application. Oso doesn't manage authorization data. The data
+stays in your application's existing data store.
+
+All the data we've defined so far in the policy is static: it isn't
+changed by end users of the application. The development team
+modifies permission associations with roles, and the list of roles for
+each resource by updating the policy. But, some parts of this policy
+must be dynamic: the association of users with a role.
+
+Write a `has_role` rule to tell Oso whether your users have a particular
+role:
+
+```polar
+has_role(user: User, role_name: String, repo: Repository) if
+  role in actor.roles and
+  role_name = role.name and
+  repository = role.repository;
+```
+
+{{% minicallout %}}
+This is an example of a full Polar rule. We'll go more into writing
+rules in the [Write Polar rules](write-rules).
+{{% /minicallout %}}
+
+The `has_role` rule uses the user object passed into Oso by your
+application to lookup roles. In this example, Polar will access the
+`roles` field on the `user` object and look up the role names that
+the user has. Here's an example {{% lang %}} data model implemented using
+dataclasses that could be used by this rule. In your application, you'll
+likely use your existing User model to maintain this information.
+
+{{< literalInclude
+    path="examples/add-to-your-application/python/app/models.py"
+    from="# docs: start"
+    to="# docs: end" >}}
+
+### Allowing access
+
+Oso policies have a special rule: the `allow` rule. The `allow` rule is
+the entrypoint to the policy, and is used by the Oso library to enforce
```

Comment:
Not sure that's quite right, because any rule can be used by the library to check if a user can perform an action on a resource. The allow rules is special because it's the first rule that's queried, or the entrypoint.